server/src/bin/ingestion-worker.rs

```diff
@@ -333,25 +321,233 @@ async fn ingestion_worker(
     }
 }
 
-#[tracing::instrument(skip(payload, web_pool, dataset_config))]
+pub async fn bulk_upload_chunks(
+    payload: BulkUploadIngestionMessage,
+    web_pool: actix_web::web::Data<models::Pool>,
+) -> Result<Vec<uuid::Uuid>, ServiceError> {
+    let dataset_config = payload.dataset_configuration;
+
+    // Being blocked out because it is difficult to create multiple split_avg embeddings in batch
+    let split_average_being_used = payload
+        .ingestion_messages
+        .iter()
+        .any(|message| message.chunk.split_avg.unwrap_or(false));
+
+    // Being blocked out because it adds complications with
+    // Duplicates being filtered out and trying to relate
+    // them back to the original messages, especially dangerous
+    // since chunk_id/tracking_id might change if there is some duplicate update logic
+    let raw_vectors_being_used = payload
+        .ingestion_messages
+        .iter()
+        .any(|message| message.chunk.chunk_vector.is_some());
+
+    let upsert_by_tracking_id_being_used = payload
+        .ingestion_messages
+        .iter()
+        .any(|message| message.upsert_by_tracking_id);
+
+    if raw_vectors_being_used
+        && split_average_being_used
+        && dataset_config.DUPLICATE_DISTANCE_THRESHOLD < 1.0
+        && dataset_config.COLLISIONS_ENABLED
+        && upsert_by_tracking_id_being_used
+    {
+        let mut chunk_ids = vec![];
+        // Split average or Collisions
+        for message in payload.ingestion_messages {
+            let upload_chunk_result = upload_chunk(message, web_pool.clone()).await;
+
+            if let Ok(chunk_uuid) = upload_chunk_result {
+                chunk_ids.push(chunk_uuid);
+            }
+        }
+        return Ok(chunk_ids);
+    }
```

Comment:
Should this not be ors between the conditions?

Reply:
Good catch!, will update.